server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -281,6 +281,13 @@ public class SearchModule {
         Setting.Property.NodeScope
     );
 
+    public static final Setting<Boolean> INDICES_CONCURRENT_COLLECTION_ENABLED = Setting.boolSetting(
+        "indices.concurrent_collection_enabled",
```

Comment:
Happy to change and will do so for now, though I though about something generic like this for a while and discarded it because "concurrency" in search is pretty general. e.g. if we also go into intra-segment concurrency later, will that be the same setting or something else? 
I will make the change though for now.

Reply:
I think it's fine, and it's an escape hatch.